src/soft-wraps-transform.coffee

```diff
@@ -19,15 +22,33 @@ class SoftWrapsTransform
       if /\s/.test(@source.characterAt(sourcePosition))
         sourceEndPosition = nextSourcePosition
       else if width > @clientWidth
-        targetTraversal = Point(1, 0)
-        break
+        width = 0
+        sourceTraversal = sourceEndPosition.traversalFrom(start)
+        regions.push new Region(sourceTraversal, Point(1, 0))
+        start = sourceEndPosition
+        if @hasTabAt(sourceStartPosition)
+          # TODO: replace "2" with the real indentation
+          regions.push new Region(Point(0, 0), Point(0, 2), 'exclusive')
+          width += @baseCharacterWidth * 2
 
       sourcePosition = nextSourcePosition
 
     sourceEndPosition ?= sourcePosition
-    sourceTraversal = sourceEndPosition.traversalFrom(sourceStartPosition)
-    targetTraversal ?= sourceTraversal
-    new Region(sourceTraversal, targetTraversal)
-
-  getContent: ({sourceStartPosition, sourceEndPosition}) ->
-    @source.slice(sourceStartPosition, sourceEndPosition)
+    sourceTraversal = sourceEndPosition.traversalFrom(start)
+    regions.push(new Region(sourceTraversal, sourceTraversal)) if regions.length == 0
+    regions
+
+  hasTabAt: (position) ->
```

Comment:
By the time we get to the soft-wraps transform, we should have already expanded tabs. I think it makes more sense to compute the indentation by counting the number of whitespace characters at the beginning of the source line and just continuing that on subsequent wrapped lines, rather than worrying about the tab character directly.